src/figuresClasses.ts

```diff
@@ -2,10 +1,83 @@
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: 'triangle' = 'triangle';
+
+  readonly color: 'red' | 'green' | 'blue';
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    public readonly a: number,
+    public readonly b: number,
+    public readonly c: number,
+  ) {
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('this is very bad trianlge');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('this is very bad triangle');
```

Comment:
The error message 'this is very bad triangle' could be improved for clarity. Consider changing it to 'Invalid triangle dimensions'.